src/ast/ast.h

```diff
@@ -5,6 +5,7 @@
 #include <map>
 #include <string>
 #include <vector>
+#include <set>
```

Comment:
Prefer unordered_set. It's faster and I don't think we need the ordering, right?
```suggestion
#include <unordered_set>
```

Reply:
right, no order is needed here